test/lint/lint-git-commit-check.sh

```diff
@@ -23,10 +23,18 @@ while getopts "?" opt; do
   esac
 done
 
+# TRAVIS_BRANCH will be present in a Travis environment. For builds triggered
+# by a pull request this is the name of the branch targeted by the pull request.
+# https://docs.travis-ci.com/user/environment-variables/
+if [ -n "${TRAVIS_BRANCH}" ]; then
+  COMMIT_RANGE="$TRAVIS_BRANCH..HEAD"
+fi
+
 if [ -z "${COMMIT_RANGE}" ]; then
     if [ -n "$1" ]; then
       COMMIT_RANGE="HEAD~$1...HEAD"
     else
+      # This assumes that the target branch of the pull request will be master.
```

Comment:
Why is this a reasonable assumption?

Reply:
I assume this will mostly be used when people run the linters locally after they have worked on a pull request and before they push and open the PR (this is what I do). And I assume most people work on PRs against master most of the time and those that don't probably are not at a stage where they care about linters. If not I can't think of a better way that just works for everyone without passing in more parameters. And that is already possible with COMMIT_RANGE which I assume people will use if they work against a different branch and want to run the linter. If this would just be set to "HEAD" here for example as the fallback then the linter would run through all commits in history and find some old commits that violate this rule. We can think of other fallbacks but this one seems most reasonable.

But also this was the way it already worked before I first touched this file in my earlier change, so I did not overthink it TBH: https://github.com/bitcoin/bitcoin/pull/19654/commits/72351784b3df21a89f79076f4b814a6e700b6469

Reply:
I see, this isn't used in any automated CI.

Reply:
That's my understanding, that the linters only run on Travis.